src/main/java/sx/blah/discord/api/internal/DiscordUtils.java

```diff
@@ -538,8 +538,9 @@ public static IChannel getChannelFromJSON(IShard shard, IGuild guild, ChannelObj
 				channel = new PrivateChannel(client, recipient, id);
 			}
 		} else if (json.type == ChannelObject.Type.GUILD_TEXT || json.type == ChannelObject.Type.GUILD_VOICE) {
-			Pair<Cache<IChannel.PermissionOverride>, Cache<IChannel.PermissionOverride>> overrides =
+			Pair<Cache<PermissionOverride>, Cache<PermissionOverride>> overrides =
 					getPermissionOverwritesFromJSONs(client, json.permission_overwrites);
+			Long categoryId = json.parent_id == null ? null : Long.parseUnsignedLong(json.parent_id);
```

Comment:
You could probably avoid unnecessary boxing by using a magic number like `0L` instead.